### PR TITLE
A4A: Add Sidebar navigation follow up PR.

### PR DESCRIPTION
**Pull request by jkguidaven**
This is a follow-up PR for #87450, which was reverted due to a bug. The issue was caused by misconfiguration, resulting in some route spillover in Wordpress.com. This PR fixes the configuration to avoid the issue.

Screenshot 2024-02-15 at 7 40 33 PM 
Screenshot 2024-02-15 at 7 40 44 PM

## Proposed Changes

* Changes from #87450.
* Disable sections related to the A4A environment in Calypso Blue.

## Testing Instructions

* Please refer to the testing instruction in #87450 
* Please also test if the issue reported before no longer occurs. p1707999443449699-slack-C03NLNTPZ2T

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/controller.tsx [hunk 1]
```diff
@@ -1,9 +1,75 @@
...
 
-export const a8cForAgenciesContext: Callback = ( context, next ) => {
-	context.header = <div>Header</div>;
-	context.secondary = <div>Secondary</div>;
-	context.primary = <div>Hello, World!</div>;
+export const redirectToOverviewContext: Callback = () => {
+	page( '/overview' );
+};
+
+export const overviewContext: Callback = ( context, next ) => {
```
Comment (jkguidaven) [inline client/a8c-for-agencies/controller.tsx:9]: These are temporary codes for demonstration purposes. Please ignore if the props are hardcoded.

File: config/_shared.json [hunk 2]
```diff
@@ -156,6 +156,8 @@
 	"push_notification_vapid_key": "BDQlTOrVTrxRM0i-gCsrxjxOyul281c5WVdfstXaqZqTSpjSI9M-E99CvwwJRWDkZ0goa8VSqUCgwiexxcDuXCs",
 	"enable_all_sections": true,
 	"sections": {
+		"a8c-for-agencies": false,
```
Comment (jkguidaven) [inline config/_shared.json:159]: These are the changes the fixes the misconfiguration.

**Comment by tyxla** [comment]
Wrapping up some loose ends today, so won't have a chance to review this one, and I'm AFK all of next week. Perhaps @simison could take a quick look with regards to the `/sites` bug which was the primary cause of the revert?

**Comment by simison** [comment]
Same situation for me today. 😓 I trust the team can navigate the bug also on your own in terms of thorough testing across calypso/a4a/jetpack cloud and the /sites toggle. Check also routes for customers without sites.

**Comment by jkguidaven** [comment]
No problem at all :). I added everyone just to let you know of the changes, but the review is totally optional for the rest.

**Review by vitozev — APPROVED** [review]
Looks good to me! I can't reproduce the issue we had on production anymore, and I did extensive testing on a few different flows like:
- Accessing /sites directly without having the `Show me all my sites when logging in to WordPress.com` setting turned on
- Accessing /sites with the `Show me all my sites when logging in to WordPress.com` setting turned on
- Accessing /sites without any site.
- Accessing /overview on Jetpack Cloud.

**Pull request merged**